source/Parallel.h

```diff
@@ -0,0 +1,127 @@
+/* Parallel.h
+Copyright (c) 2024 by tibetiroka
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef PARALLEL_H_
+#define PARALLEL_H_
+
+#include <version>
+
+// Only use this header if the underlying system doesn't support the standard algorithms.
+#if !defined(__cpp_lib_parallel_algorithm) && !defined(__cpp_lib_execution)
+
+#define ES_PARALLEL_USE_TASK_QUEUE
+
+#include "TaskQueue.h"
+
+#include <algorithm>
+#include <utility>
+
+// Dummy for std::execution.
+enum class execution
+{
+	seq, par, par_unseq
+};
+
+
+
+// Holds a TaskQueue to void adding globals to the namespace.
+class Parallel
+{
+public:
+	template<class RandomIt, class Func>
+	static void RunBulk(const RandomIt begin, const RandomIt end, Func &&f);
+	static inline void Run(std::function<void()> f) {queue.Run(std::move(f));}
+	static inline void Wait() {queue.Wait();}
+private:
+	static TaskQueue queue;
+};
+
+
+
+template<class RandomIt, class Func>
+void Parallel::RunBulk(const RandomIt begin, const RandomIt end, Func &&f)
```

Comment:
```suggestion
void Parallel::RunBulk(const RandomIt begin, const RandomIt end, Func &&f)
```
a little more descriptive than just f maybe? how about func

Reply:
I'll just add a comment.